### PR TITLE
Install Nvidia Daemonset in test harness for GCE

### PR DESCRIPTION
**Pull request by dims**
This PR resurrects some of the code we dropped when we cleaned up the GCE cloud provider. In this PR, just like we had in previous versions of k8s, we load up  the manifest from`"test/e2e/testing-manifests/scheduling/nvidia-driver-installer.yaml"` and use that to install the nvidia daemonset. 

Note that this manifest is GCE only, so for AWS we still rely on provider-aws-test-infra's `kubetest2 ec2`'s support to load up the nvidia driver:
https://github.com/search?q=repo%3Akubernetes-sigs%2Fprovider-aws-test-infra%20nvidia&type=code

(We will have to revisit this situation and see if we should leave things as is or reconcile testing on both clouds to use the same approach)

xref: https://github.com/kubernetes/kubernetes/issues/124950
xref: https://github.com/kubernetes/kubernetes/pull/127423

Source of resurrected code:
https://github.com/kubernetes/kubernetes/blob/release-1.30/test/e2e/scheduling/nvidia-gpus.go

#### What type of PR is this?

/kind cleanup
/kind regression

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
we have a 🟢 run - https://testgrid.k8s.io/presubmits-kubernetes-nonblocking#pull-kubernetes-e2e-gce-device-plugin-gpu&width=20

File: test/e2e/node/gpu.go [hunk 1]
```diff
@@ -204,3 +224,83 @@ func checkEnvironmentAndSkipIfNeeded(ctx context.Context, clientSet clientset.In
...
+	nodeList, err := f.ClientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	framework.ExpectNoError(err, "getting node list")
+	for _, node := range nodeList.Items {
+		framework.Logf("Nodename: %v, OS Image: %v", node.Name, node.Status.NodeInfo.OSImage)
+	}
+}
+
+const (
+	// Nvidia driver installation can take upwards of 5 minutes.
+	driverInstallTimeout = 10 * time.Minute
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:256]: wondering if we want this get through env var ? Do you see the need for this being dynamic  ? Like - 
```
timeout := framework.GetEnvOrDefault("DRIVER_INSTALL_TIMEOUT", 10*time.Minute)``` ?

File: test/e2e/node/gpu.go [hunk 2]
```diff
@@ -204,3 +224,83 @@ func checkEnvironmentAndSkipIfNeeded(ctx context.Context, clientSet clientset.In
...
+)
+
+// SetupNVIDIAGPUNode install Nvidia Drivers and wait for Nvidia GPUs to be available on nodes
+func SetupNVIDIAGPUNode(ctx context.Context, f *framework.Framework) *e2edebug.ContainerResourceGatherer {
+	logOSImages(ctx, f)
```
Comment (hakuna-matatah) [inline test/e2e/node/gpu.go:261]: May be we could run this in go routine as future optimization ? and continue with rest ? The lesser the amount of time for this test, better it is for cost purposes ?

**Review by hakuna-matatah — COMMENTED** [review]
/lgtm

overall looks fine to me. Thank you.

File: test/e2e/node/gpu.go (hunk 1 above)
Comment (dims) [reply test/e2e/node/gpu.go:256]: want to leave this as is to preserve parity with older versions (can't cherry pick with new "features")

File: test/e2e/node/gpu.go (hunk 2 above)
Comment (dims) [reply test/e2e/node/gpu.go:261]: we can't fun the test until the `nvidia.com/gpu` shows up in `capacity` and `allocatable` fields. this test is run only in a handful of limited jobs that are dedicated to GPU testing, so we should be fine. (also don't want to make it more complex so would be harder to backport)

**Comment by dims** [comment]
@hakuna-matatah thanks for the review, let me get the CI go green and then will poke you for `/lgtm`

**Comment by dims** [comment]
we have a green run again - https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127456/pull-kubernetes-e2e-gce-device-plugin-gpu/1836556273802285056

@hakuna-matatah please recast your `/lgtm`

**Pull request merged**